src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp

```diff
@@ -69,25 +69,32 @@ class BiotPorosityUpdates : public PorosityBaseUpdates
   real64 dGrainDensity_dPressure() const { return 1.0 / m_grainBulkModulus; }
 
   GEOSX_HOST_DEVICE
-  void updateFromPressureAndStrain( localIndex const k,
-                                    localIndex const q,
-                                    real64 const & deltaPressure,
-                                    real64 const (&strainIncrement)[6],
-                                    real64 & dPorosity_dPressure,
-                                    real64 & dPorosity_dVolStrain ) const
+  void updateFromPressureTemperatureAndStrain( localIndex const k,
+                                               localIndex const q,
+                                               real64 const & deltaPressure,
+                                               real64 const & deltaTemperature,
+                                               real64 const (&strainIncrement)[6],
+                                               real64 const & thermalExpansionCoefficient,
+                                               real64 & dPorosity_dVolStrain,
+                                               real64 & dPorosity_dPressure,
+                                               real64 & dPorosity_dTemperature ) const
   {
     real64 const biotSkeletonModulusInverse = (m_biotCoefficient[k] - m_referencePorosity[k]) / m_grainBulkModulus;
+    real64 const porosityThermalExpansion = 3 * thermalExpansionCoefficient * m_biotCoefficient[k];
```

Comment:
do you want to use directly m_thermalExpansionCoefficient here as you are using m_biotCoefficient, m_referencePorosity, etc.

Reply:
Ideally, we would want the `thermalExpansionCoefficient` to live in one place (i.e., not duplicated), and I think it will make sense to have it in the solid model (see what I have done in #2195). That is why it is treated differently from the Biot coefficient and the reference porosity.

So in #2195, in the `PorousSolid` constitutive model, I get the `thermalExpansionCoefficient` from the solid model, and I pass it to the porosity model as an argument of the `updateFromPressureTemperatureAndStrain` function. I think it is cleaner like this.